### PR TITLE
Do not load database in string, use (string|struct).unpack, misc fixes

### LINKED ISSUE #3: Use new 5.3 string.unpack instead of home grown reading.
At the moment we use some hacky routines to read/write doubles: https://github.com/daurnimator/mmdblua/blob/f0644b7478ec91fd8a7d85a9f0f259f444ba20cd/IEEE-754.lua
As well as some simple routines to read ints: https://github.com/daurnimator/mmdblua/blob/f0644b7478ec91fd8a7d85a9f0f259f444ba20cd/mmdb.lua#L179

Lua 5.3 brings with it new function [`string.unpack`](http://www.lua.org/manual/5.3/manual.html#pdf-string.unpack) which can be used to tidy this up.
[compat-5.3](https://github.com/keplerproject/lua-compat-5.3) can be used to bring this functionality to lua 5.1 and 5.2. Preferably we stay compatible with [Roberto's struct library](http://www.inf.puc-rio.br/~roberto/struct/) too.

### PR DESCRIPTION
**Pull request by pisto**
Fixes #3 and #4 . I am testing the changes on my services, wait a day or two and I'll report results.

### CODE REVIEW

**Comment by daurnimator** [comment]
I already merged the first two commits as they're trivial.

File: mmdb.lua [hunk 1]
```diff
@@ -1,32 +1,63 @@
...
+function file_contents_methods:byte ( i , j )
+	return self:sub ( i , j or i ) :byte ( 1 , -1 )
+end
+
+local function open_db ( filename , read_once )
 	local fd = assert ( io.open ( filename , "rb" ) )
-	local contents = assert ( fd:read ( "*a" ) )
+	local size, contents
+	if read_once then
+    size = #contents
```
Comment (daurnimator) [inline mmdb.lua]: indentation

**Comment by pisto** [comment]
Should have taken care of all the remarks.

**Comment by pisto** [comment]
By the way are you sure you want these source files to be in crlf format?

**Comment by daurnimator** [comment]
> By the way are you sure you want these source files to be in crlf format?

No; should just be a LF.

**Comment by daurnimator** [comment]
Sorry, but after reflecting on this a while, I realised that seeking a file here is *not* the correct answer: it will result in un-necessary slowness due to `seek` syscall overhead.

What I suggest instead is an ffi-only optimisation that uses an `mmap` call (perhaps via ljsyscall).

**Pull request closed without merging**